### PR TITLE
libphonenumber: 8.9.9 -> 8.10.17

### PR DESCRIPTION
**Pull request by dtzWill**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/libraries/libphonenumber/default.nix [hunk 1]
```diff
@@ -1,7 +1,7 @@
 { stdenv, fetchFromGitHub, cmake, gmock, boost, pkgconfig, protobuf, icu }:
 
 let
-  version = "8.9.9";
+  version = "8.10.17";
 in
 stdenv.mkDerivation {
   name = "phonenumber-${version}";
```
Comment (jonringer) [inline pkgs/development/libraries/libphonenumber/default.nix]: while we're at it
```suggestion
  pname = "phonenumber";
  version = "8.10.17";
```

File: pkgs/development/libraries/libphonenumber/default.nix [hunk 2]
```diff
@@ -1,17 +1,17 @@
 { stdenv, fetchFromGitHub, cmake, gmock, boost, pkgconfig, protobuf, icu }:
 
 let
-  version = "8.9.9";
+  version = "8.10.17";
```
Comment (jonringer) [inline pkgs/development/libraries/libphonenumber/default.nix]: Just my preference, but I would move the version attr into the derivation
```suggestion
```

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM (some suggestions)
leaf package

```
[3 built, 15 copied (226.5 MiB), 27.2 MiB DL]
https://github.com/NixOS/nixpkgs/pull/68067
1 package were build:
libphonenumber
```

I think the darwin build might work if you did something like:
```
buildInputs = [...]
++ stdenv.lib.optionals stdenv.isDarwin [ libiconv CoreFoundation ];
```
seems to not be able to link against corefoundation

**Comment by JohnAZoidberg** [comment]
We've got 8.10.20 since ##70496, even with the additional proposed changes.

**Pull request closed without merging**